### ISSUE TITLE
Perform Hyperparameter Tuning using GridSearchCV 

### ISSUE THREAD

**Issue by Priyanka142806** [DSC-CETB/HR_Analytics_CaseStudy#2]
The performance of a model significantly depends on the value of hyperparameters. Use GridSearchCV to know the optimal/best hyperparameters.

**ISSUE**
 The performance of a model significantly depends on the value of hyperparameters. There is no way to know the best values for hyperparameters. So, we need to try all possible values to know the optimal/best hyperparameters so as to improve the accuracy. Doing this manually could take a considerable amount of time and resources

**SOLUTION**
**Using GridSearchCV** to tune the model would help us automate this process. GridSearchCV is a function that comes in Scikit-learn’s(or SK-learn) model_selection package
A range of values for each hyperparameters to the GridSearchCV function is passed by defining a dictionary  in which a particular hyperparameter along with the values it can take in a list is mentioned.
GridSearchCV tries all the combinations of the values passed in the dictionary and evaluates the model for each combination using the Cross-Validation method. Hence after using this function we get accuracy/loss for every combination of hyperparameters and we can choose the one with the best performance.

**![Screenshot (206)](https://user-images.githubusercontent.com/66913821/136989410-cb7cacae-627c-4f19-aa92-34ba62143c21.png)**
